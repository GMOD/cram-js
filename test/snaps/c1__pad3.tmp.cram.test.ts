import { testFile } from '../testUtil'
testFile('c1#pad3.tmp.cram')
