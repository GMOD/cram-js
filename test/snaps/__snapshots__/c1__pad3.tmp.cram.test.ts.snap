// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can dump the whole c1#pad3.tmp.cram without error 1`] = `
[
  {
    "fileId": "-",
    "magic": "CRAM",
    "majorVersion": 3,
    "minorVersion": 0,
  },
  {
    "containerHeader": {
      "_endPosition": 45,
      "_size": 19,
      "alignmentSpan": 0,
      "crc32": 855301221,
      "landmarks": [
        0,
        152,
      ],
      "length": 237,
      "numBases": 0,
      "numBlocks": 2,
      "numLandmarks": 2,
      "numRecords": 0,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 0,
    },
    "data": [
      {
        "_endPosition": 45,
        "_size": 19,
        "alignmentSpan": 0,
        "crc32": 855301221,
        "landmarks": [
          0,
          152,
        ],
        "length": 237,
        "numBases": 0,
        "numBlocks": 2,
        "numLandmarks": 2,
        "numRecords": 0,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 0,
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "c1",
              },
              {
                "tag": "LN",
                "value": "10",
              },
              {
                "tag": "M5",
                "value": "e224bd8a70e5466c6a033be886045c85",
              },
              {
                "tag": "UR",
                "value": "/Users/rbuels/dev/htslib/test/c1.fa",
              },
            ],
            "tag": "SQ",
          },
          {
            "data": [
              {
                "tag": "ID",
                "value": "p.sam",
              },
              {
                "tag": "SM",
                "value": "unknown",
              },
              {
                "tag": "LB",
                "value": "p.sam",
              },
            ],
            "tag": "RG",
          },
        ],
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "c1",
              },
              {
                "tag": "LN",
                "value": "10",
              },
              {
                "tag": "M5",
                "value": "e224bd8a70e5466c6a033be886045c85",
              },
              {
                "tag": "UR",
                "value": "/Users/rbuels/dev/htslib/test/c1.fa",
              },
            ],
            "tag": "SQ",
          },
          {
            "data": [
              {
                "tag": "ID",
                "value": "p.sam",
              },
              {
                "tag": "SM",
                "value": "unknown",
              },
              {
                "tag": "LB",
                "value": "p.sam",
              },
            ],
            "tag": "RG",
          },
        ],
      },
    ],
  },
  {
    "compressionScheme": {
      "APdelta": true,
      "dataSeriesEncoding": {
        "AP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "BB": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 42,
              },
              "parametersBytes": 1,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 37,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 6,
        },
        "BF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "CF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              3,
            ],
          },
          "parametersBytes": 4,
        },
        "DL": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 29,
          },
          "parametersBytes": 1,
        },
        "FC": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 27,
          },
          "parametersBytes": 1,
        },
        "FN": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 26,
          },
          "parametersBytes": 1,
        },
        "FP": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 28,
          },
          "parametersBytes": 1,
        },
        "IN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 13,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "MF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "MQ": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "NP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "NS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "PD": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 35,
          },
          "parametersBytes": 1,
        },
        "QS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 12,
          },
          "parametersBytes": 1,
        },
        "RG": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "RI": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "RL": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 25,
          },
          "parametersBytes": 1,
        },
        "RN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 11,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "SC": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 14,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "TL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "TS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
      },
      "readNamesIncluded": true,
      "referenceRequired": false,
      "substitutionMatrix": [
        [
          "C",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "T",
        ],
      ],
      "tagEncoding": {},
      "tagIdsDictionary": [
        [],
      ],
    },
    "containerHeader": {
      "_endPosition": 302,
      "_size": 20,
      "alignmentSpan": 5,
      "crc32": 2515036048,
      "landmarks": [
        190,
        569,
      ],
      "length": 984,
      "numBases": 126,
      "numBlocks": 22,
      "numLandmarks": 2,
      "numRecords": 12,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 6,
    },
    "data": [
      {
        "_endPosition": 302,
        "_size": 20,
        "alignmentSpan": 5,
        "crc32": 2515036048,
        "landmarks": [
          190,
          569,
        ],
        "length": 984,
        "numBases": 126,
        "numBlocks": 22,
        "numLandmarks": 2,
        "numRecords": 12,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 6,
      },
      {
        "features": [
          {
            "alignmentStart": 6,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 5,
            "mappingQuality": 0,
            "qualityScores": [
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "I",
                "data": "AACCG",
                "pos": 1,
                "refPos": 6,
              },
              {
                "code": "P",
                "data": 6,
                "pos": 6,
                "refPos": 6,
              },
            ],
            "readGroupId": 0,
            "readLength": 10,
            "readName": "s0a",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 498,
          },
          {
            "alignmentStart": 6,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 5,
            "mappingQuality": 0,
            "qualityScores": [
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "I",
                "data": "AACCG",
                "pos": 1,
                "refPos": 6,
              },
              {
                "code": "P",
                "data": 6,
                "pos": 6,
                "refPos": 6,
              },
            ],
            "readGroupId": 0,
            "readLength": 10,
            "readName": "s0b",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 499,
          },
          {
            "alignmentStart": 6,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 5,
            "mappingQuality": 0,
            "qualityScores": [
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "I",
                "data": "AACCG",
                "pos": 1,
                "refPos": 6,
              },
              {
                "code": "P",
                "data": 6,
                "pos": 6,
                "refPos": 6,
              },
            ],
            "readGroupId": 0,
            "readLength": 10,
            "readName": "s0c",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 500,
          },
          {
            "alignmentStart": 6,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 5,
            "mappingQuality": 0,
            "qualityScores": [
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "I",
                "data": "AACCG",
                "pos": 1,
                "refPos": 6,
              },
              {
                "code": "P",
                "data": 6,
                "pos": 6,
                "refPos": 6,
              },
            ],
            "readGroupId": 0,
            "readLength": 10,
            "readName": "s0d",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 501,
          },
          {
            "alignmentStart": 6,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 5,
            "mappingQuality": 0,
            "qualityScores": [
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "I",
                "data": "AACCGGTTAAC",
                "pos": 1,
                "refPos": 6,
              },
            ],
            "readGroupId": 0,
            "readLength": 16,
            "readName": "s1",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 502,
          },
          {
            "alignmentStart": 6,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 5,
            "mappingQuality": 0,
            "qualityScores": [
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "I",
                "data": "AACCG",
                "pos": 1,
                "refPos": 6,
              },
              {
                "code": "P",
                "data": 1,
                "pos": 6,
                "refPos": 6,
              },
              {
                "code": "I",
                "data": "TTAA",
                "pos": 6,
                "refPos": 6,
              },
              {
                "code": "P",
                "data": 1,
                "pos": 10,
                "refPos": 6,
              },
            ],
            "readGroupId": 0,
            "readLength": 14,
            "readName": "s2",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 503,
          },
          {
            "alignmentStart": 6,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 5,
            "mappingQuality": 0,
            "qualityScores": [
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "I",
                "data": "AACCGGTT",
                "pos": 1,
                "refPos": 6,
              },
              {
                "code": "P",
                "data": 3,
                "pos": 9,
                "refPos": 6,
              },
            ],
            "readGroupId": 0,
            "readLength": 13,
            "readName": "s3",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 504,
          },
        ],
        "header": {
          "_endPosition": 537,
          "_size": 40,
          "compressedSize": 36,
          "compressionMethod": "raw",
          "content": Uint8Array [
            0,
            6,
            5,
            7,
            0,
            9,
            8,
            11,
            12,
            13,
            25,
            26,
            27,
            28,
            35,
            255,
            255,
            255,
            255,
            15,
            153,
            15,
            79,
            183,
            231,
            205,
            24,
            189,
            209,
            25,
            192,
            96,
            167,
            121,
            160,
            184,
          ],
          "contentId": 0,
          "contentPosition": 497,
          "contentType": "MAPPED_SLICE_HEADER",
          "crc32": 1750556365,
          "parsedContent": {
            "_endPosition": 338,
            "_size": 36,
            "contentIds": [
              11,
              12,
              13,
              25,
              26,
              27,
              28,
              35,
            ],
            "md5": [
              153,
              15,
              79,
              183,
              231,
              205,
              24,
              189,
              209,
              25,
              192,
              96,
              167,
              121,
              160,
              184,
            ],
            "numBlocks": 9,
            "numContentIds": 8,
            "numRecords": 7,
            "recordCounter": 0,
            "refBaseBlockId": -1,
            "refSeqId": 0,
            "refSeqSpan": 5,
            "refSeqStart": 6,
          },
          "uncompressedSize": 36,
        },
      },
      {
        "block": {
          "_endPosition": 546,
          "_size": 4,
          "compressedSize": 0,
          "compressionMethod": "raw",
          "content": Uint8Array [],
          "contentId": 0,
          "contentPosition": 542,
          "contentType": "CORE_DATA",
          "crc32": 4059825967,
          "uncompressedSize": 0,
        },
      },
      {
        "block": {
          "_endPosition": 600,
          "_size": 49,
          "compressedSize": 45,
          "compressionMethod": "gzip",
          "content": Uint8Array [
            115,
            48,
            97,
            0,
            115,
            48,
            98,
            0,
            115,
            48,
            99,
            0,
            115,
            48,
            100,
            0,
            115,
            49,
            0,
            115,
            50,
            0,
            115,
            51,
            0,
          ],
          "contentId": 11,
          "contentPosition": 551,
          "contentType": "EXTERNAL_DATA",
          "crc32": 1830714932,
          "uncompressedSize": 25,
        },
      },
      {
        "block": {
          "_endPosition": 632,
          "_size": 27,
          "compressedSize": 23,
          "compressionMethod": "gzip",
          "content": Uint8Array [
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
            255,
          ],
          "contentId": 12,
          "contentPosition": 605,
          "contentType": "EXTERNAL_DATA",
          "crc32": 1514207066,
          "uncompressedSize": 83,
        },
      },
      {
        "block": {
          "_endPosition": 684,
          "_size": 47,
          "compressedSize": 43,
          "compressionMethod": "gzip",
          "content": Uint8Array [
            65,
            65,
            67,
            67,
            71,
            0,
            65,
            65,
            67,
            67,
            71,
            0,
            65,
            65,
            67,
            67,
            71,
            0,
            65,
            65,
            67,
            67,
            71,
            0,
            65,
            65,
            67,
            67,
            71,
            71,
            84,
            84,
            65,
            65,
            67,
            0,
            65,
            65,
            67,
            67,
            71,
            0,
            84,
            84,
            65,
            65,
            0,
            65,
            65,
            67,
            67,
            71,
            71,
            84,
            84,
            0,
          ],
          "contentId": 13,
          "contentPosition": 637,
          "contentType": "EXTERNAL_DATA",
          "crc32": 2873832221,
          "uncompressedSize": 56,
        },
      },
      {
        "block": {
          "_endPosition": 718,
          "_size": 29,
          "compressedSize": 25,
          "compressionMethod": "gzip",
          "content": Uint8Array [
            10,
            10,
            10,
            10,
            16,
            14,
            13,
          ],
          "contentId": 25,
          "contentPosition": 689,
          "contentType": "EXTERNAL_DATA",
          "crc32": 1188711010,
          "uncompressedSize": 7,
        },
      },
      {
        "block": {
          "_endPosition": 752,
          "_size": 29,
          "compressedSize": 25,
          "compressionMethod": "gzip",
          "content": Uint8Array [
            2,
            2,
            2,
            2,
            1,
            4,
            2,
          ],
          "contentId": 26,
          "contentPosition": 723,
          "contentType": "EXTERNAL_DATA",
          "crc32": 2424666958,
          "uncompressedSize": 7,
        },
      },
      {
        "block": {
          "_endPosition": 793,
          "_size": 36,
          "compressedSize": 32,
          "compressionMethod": "rans",
          "content": Uint8Array [
            73,
            80,
            73,
            80,
            73,
            80,
            73,
            80,
            73,
            73,
            80,
            73,
            80,
            73,
            80,
          ],
          "contentId": 27,
          "contentPosition": 757,
          "contentType": "EXTERNAL_DATA",
          "crc32": 3842539510,
          "uncompressedSize": 15,
        },
      },
      {
        "block": {
          "_endPosition": 837,
          "_size": 39,
          "compressedSize": 35,
          "compressionMethod": "gzip",
          "content": Uint8Array [
            1,
            5,
            1,
            5,
            1,
            5,
            1,
            5,
            1,
            1,
            5,
            0,
            4,
            1,
            8,
          ],
          "contentId": 28,
          "contentPosition": 798,
          "contentType": "EXTERNAL_DATA",
          "crc32": 2024508679,
          "uncompressedSize": 15,
        },
      },
      {
        "block": {
          "_endPosition": 871,
          "_size": 29,
          "compressedSize": 25,
          "compressionMethod": "gzip",
          "content": Uint8Array [
            6,
            6,
            6,
            6,
            1,
            1,
            3,
          ],
          "contentId": 35,
          "contentPosition": 842,
          "contentType": "EXTERNAL_DATA",
          "crc32": 1799963567,
          "uncompressedSize": 7,
        },
      },
      {
        "features": [
          {
            "alignmentStart": 6,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 5,
            "mappingQuality": 0,
            "qualityScores": [
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "I",
                "data": "AACCG",
                "pos": 1,
                "refPos": 6,
              },
              {
                "code": "P",
                "data": 3,
                "pos": 6,
                "refPos": 6,
              },
              {
                "code": "I",
                "data": "AAC",
                "pos": 6,
                "refPos": 6,
              },
            ],
            "readGroupId": 0,
            "readLength": 13,
            "readName": "s4",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 884,
          },
          {
            "alignmentStart": 6,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 5,
            "mappingQuality": 0,
            "qualityScores": [
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "I",
                "data": "AACC",
                "pos": 1,
                "refPos": 6,
              },
              {
                "code": "P",
                "data": 3,
                "pos": 5,
                "refPos": 6,
              },
              {
                "code": "I",
                "data": "TA",
                "pos": 5,
                "refPos": 6,
              },
              {
                "code": "P",
                "data": 2,
                "pos": 7,
                "refPos": 6,
              },
              {
                "code": "D",
                "data": 1,
                "pos": 7,
                "refPos": 6,
              },
            ],
            "readGroupId": 0,
            "readLength": 10,
            "readName": "s5",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 885,
          },
          {
            "alignmentStart": 6,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 5,
            "mappingQuality": 0,
            "qualityScores": [
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "I",
                "data": "AA",
                "pos": 1,
                "refPos": 6,
              },
              {
                "code": "P",
                "data": 3,
                "pos": 3,
                "refPos": 6,
              },
              {
                "code": "I",
                "data": "GTTAAC",
                "pos": 3,
                "refPos": 6,
              },
              {
                "code": "D",
                "data": 3,
                "pos": 9,
                "refPos": 6,
              },
            ],
            "readGroupId": 0,
            "readLength": 10,
            "readName": "s6",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 886,
          },
          {
            "alignmentStart": 6,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 5,
            "mappingQuality": 0,
            "qualityScores": [
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "I",
                "data": "AACC",
                "pos": 1,
                "refPos": 6,
              },
              {
                "code": "P",
                "data": 7,
                "pos": 5,
                "refPos": 6,
              },
              {
                "code": "D",
                "data": 1,
                "pos": 5,
                "refPos": 6,
              },
            ],
            "readGroupId": 0,
            "readLength": 8,
            "readName": "s7",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 887,
          },
          {
            "alignmentStart": 6,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 0,
            "mappingQuality": 0,
            "qualityScores": [
              0,
              0,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "P",
                "data": 7,
                "pos": 1,
                "refPos": 6,
              },
              {
                "code": "I",
                "data": "TA",
                "pos": 1,
                "refPos": 6,
              },
              {
                "code": "P",
                "data": 2,
                "pos": 3,
                "refPos": 6,
              },
            ],
            "readGroupId": 0,
            "readLength": 2,
            "readName": "s8",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 888,
          },
        ],
        "header": {
          "_endPosition": 917,
          "_size": 41,
          "compressedSize": 37,
          "compressionMethod": "raw",
          "content": Uint8Array [
            0,
            6,
            5,
            5,
            7,
            10,
            9,
            11,
            12,
            13,
            25,
            26,
            27,
            28,
            29,
            35,
            255,
            255,
            255,
            255,
            15,
            153,
            15,
            79,
            183,
            231,
            205,
            24,
            189,
            209,
            25,
            192,
            96,
            167,
            121,
            160,
            184,
          ],
          "contentId": 0,
          "contentPosition": 876,
          "contentType": "MAPPED_SLICE_HEADER",
          "crc32": 1865669966,
          "parsedContent": {
            "_endPosition": 339,
            "_size": 37,
            "contentIds": [
              11,
              12,
              13,
              25,
              26,
              27,
              28,
              29,
              35,
            ],
            "md5": [
              153,
              15,
              79,
              183,
              231,
              205,
              24,
              189,
              209,
              25,
              192,
              96,
              167,
              121,
              160,
              184,
            ],
            "numBlocks": 10,
            "numContentIds": 9,
            "numRecords": 5,
            "recordCounter": 7,
            "refBaseBlockId": -1,
            "refSeqId": 0,
            "refSeqSpan": 5,
            "refSeqStart": 6,
          },
          "uncompressedSize": 37,
        },
      },
    ],
  },
]
`;
